docs/source/index.rst

```diff
@@ -1,20 +1,204 @@
+   Since the query language always has the same semantics regardless of the underlying database,
+   we have been able to build a :doc:`Schema Stitching <advanced_features/schema_stitching>` system
+   that allows for seamless cross-database querying.
+
+Getting Started
+---------------
+
+Generating the necessary schema info
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+..  TODO: Encapsulate all schema info in a SchemaInfo class.
+
+To use the GraphQL compiler the first thing one needs to do is to generate the schema info from the
+underlying database as in the example below. Even though the example below generates schema info
+from an OrientDB database, it is meant as a generic schema info generation example.
+See the target database homepage for schema generation instructions.
+
+.. code:: python
+
+    from graphql_compiler import (
+        get_graphql_schema_from_orientdb_schema_data
+    )
+    from graphql_compiler.schema_generation.orientdb.utils import ORIENTDB_SCHEMA_RECORDS_QUERY
+
+    client = your_function_that_returns_an_orientdb_client()
+    schema_records = client.command(ORIENTDB_SCHEMA_RECORDS_QUERY)
+    schema_data = [record.oRecordData for record in schema_records]
+    schema, type_equivalence_hints = get_graphql_schema_from_orientdb_schema_data(schema_data)
+
+At the core of generated schema info is the GraphQL :code:`schema`. The database might be
+reflected in the :code:`schema` as follows:
+
+.. code::
+
+    type Animal {
+        name: String
+        out_Animal_LivesIn: [Continent]
+    }
+
+    type Continent {
+        name: String
+        in_AnimalLivesIn: [Animal]
+    }
+
+In the :code:`schema` above:
+
+-   :code:`Animal` represents a non-abstract vertex. For relational databases, we think of
+    tables as the non-abstract vertices.
+-   :code:`name` is a **property field** and :code:`out_Animal_LivesIn` is a **vertex field**,
+    which represent an outbound edge to a vertex in the graph. For graph databases,
+    edges can be reflected from the database schema. However, for relational databases edges have
+    to be manually specified. See :doc:`SQL <databases/sql>` for more information.
+
+Query Compilation and Execution
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Once we have the schema info we can write the following query to get the names of all the animals
+that live in Africa:
+
+.. code:: python
+
+    graphql_query = """
+    {
+        Animal {
+            name @output(out_name: "animal_name")
+            out_Animal_LivesIn {
+                name @filter(op_name: "=", value: ["$continent"])
+            }
+        }
+    }
+    """
+    parameters = {'continent': 'Africa'}
+
+There are a couple of things to notice about queries:
 
+- All queries start with a vertex and branch to other vertices using **vertex fields**.
+- **Directives**, such as :code:`@output` and :code:`@filter`, specify the semantics of a query.
+  :code:`@output` indicates which properties to emit and :code:`@filter` specifies a filter
```

Comment:
`emit` is not the right word here. Code is emitted, values of property fields are returned. Please rewrite for clarity. Try not to unnecessarily cram unrelated concepts into one sentence -- output and filter can be in separate sentences here.